components/nimbus/src/persistence.rs

```diff
@@ -176,6 +179,38 @@ impl SingleStore {
         }
     }
 
+    /// Fork of collect_all that simply drops records that fail to read
+    /// rather than returning an Err up the stack in lieu of any records at
+    /// all.  This likely wants to be just a parameter to collect_all, but
+    /// for now....
+    ///
+    pub fn try_collect_all<'r, T, R>(&self, reader: &'r R) -> Result<Vec<T>>
+    where
+        R: Readable<'r>,
+        T: serde::Serialize + for<'de> serde::Deserialize<'de>,
+    {
+        let mut result = Vec::new();
+        let mut iter = self.store.iter_start(reader)?;
+        while let Some(Ok((_, data))) = iter.next() {
+            if let rkv::Value::Json(data) = data {
+                let unserialized = serde_json::from_str::<T>(&data);
+                match unserialized {
+                    Ok(value) => result.push(value),
+                    Err(e) => {
+                        // If there is an error, we won't push this onto the
+                        // result Vec, but we won't blow up the entire
+                        // deserialization either.
+                        log::warn!(
+                            "try_collect_all: discarded a record while deserializing with: {:?}",
+                            e
+                        );
```

Comment:
What do you think about dropping a `log::debug!` with the contents that we couldn't deserialize?

Reply:
Assuming I stick with the current migrator strategy; I like it!

Reply:
I still need to look into this.

Reply:
Done!